### ISSUE TITLE
Bug - Regex fails on trailing dashes

### ISSUE THREAD

**Issue by JosephScript** [sereneinserenade/tiptap-search-and-replace#4]
Hey thanks for taking the time to review this issue. When using the plugin v0.0.2 I run into an error when searching with a trailing hyphens. Example when searching "co-workers" or even the word "co-":

```
Invalid regular expression: /co\-worker/giu: Invalid escape
```

Here is the code causing the problem:

```js
 const getRegex = (s, disableRegex, caseSensitive) => {
   return RegExp(disableRegex ? s.replace(/[-/\\^$*+?.()|[\]{}]/g, "\\$&") : s, caseSensitive ? "gu" : "gui");
 };
```

---

Edited to better explain the issue

**Comment by JosephScript** [comment]
This seems to work for me:

```
RegExp(disableRegex ? s.replace(/[.*+?^${}()|[\]\\]/g, '\\$&') : s, caseSensitive ? "gu" : "gui");
```

This regex accounts for all the special characters that need escaping in a regular expression context and is comprehensive enough for general use cases.

**Comment by sereneinserenade** [comment]
@JosephScript, thanks for reporting both issue & fix. I've published `0.0.3` patch version for the fix with #5.

**Comment by sereneinserenade** [comment]
@JosephScript for some reason it wasn't fixed in `0.0.3`, published `0.0.4` for the final fix.
